### ISSUE TITLE
Confirmation popup for feature areas that expose private data

### ISSUE THREAD

**Issue by thadguidry** [OpenRefine/OpenRefine#2172]
**Is your feature request related to a problem or area of OpenRefine? Please describe.**
Some features of OpenRefine can expose private data and users should be made more aware of that prior to the execution.

**Describe the solution you'd like**
 We could add a confirmation popup on features that states "This feature of OpenRefine will share parts of your data which may be private for you...Are you sure you want to continue? (yes/no/always)"

**Describe alternatives you've considered**
None

**Additional context**
This feature popped in my head after reading about user concerns on our group mailing list https://groups.google.com/d/topic/openrefine/SmanJL5X2rs/discussion

**Comment by wetneb** [comment]
I'd be reluctant to use a popup for this - this is going to be quite frustrating for anyone who uses these features often. We could add a small note in some of the dialogs that configure these features, but even so I would say that it has to be quite non-intruding and compact.

For instance I think it must be fairly obvious to many of our users that "Add column by fetching URLs" does communicate with external services. If we want to add a warning, it should be easy to discard and it would then never show up again.

**Comment by thadguidry** [comment]
Agree, that's why the "always". :)  I think the first time we want to be intruding and in-your-face to users.  After that there is an ("always" option to continue without further interruptions in the future) where it's stored in preferences, so they don't have to bother with the notice in the future of they don't want.
